### PR TITLE
fcl: init at 0.6.1

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

Adds the [Flexible Collision Library](https://github.com/flexible-collision-library/fcl) and its dependencies. This package is used by several Robot Operating System (ROS) packages, and would allow them to be included in [nix-ros-overlay](https://github.com/lopsided98/nix-ros-overlay). On the other hand, fcl is useful outside of ROS, so I thought it should belong in nixpkgs rather than the overlay.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123060` at 57af4ffe run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/eb6338ce630a8d97ba2102151497a47b">fcl</a></li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>libccd</li>
    <li>octomap</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/libraries/octomap/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+stdenv.mkDerivation rec {
+  pname = "octomap";
+  version = "1.9.6";
+
+  src = fetchFromGitHub {
+    owner = "OctoMap";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "03v341dffa0pfzmf2431xb5nq50zq9zlhgl6k2aa3fsza5xmbb70";
+  } + "/octomap";
```
Comment (fricklerhandwerk) [inline pkgs/development/libraries/octomap/default.nix:12]:
```suggestion
  };
  
  sourceRoot = "source/octomap";
```

**Review by fricklerhandwerk — CHANGES_REQUESTED** [review]
While it doesn't look as nice because [for some reason](https://github.com/NixOS/nixpkgs/issues/31172#issuecomment-596430923) we have to prepend `"source"`, this will be easier to refactor once we don't have to add the prefix any more.

**Comment by lopsided98** [comment]
I added some upstream patches to fix building on aarch64.

**Pull request merged**